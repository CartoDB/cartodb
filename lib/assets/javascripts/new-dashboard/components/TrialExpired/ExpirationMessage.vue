<template>
  <section>
    <div class="container warning-container">
      <div class="warning-icon">
        <img src="../../assets/icons/common/warning-icon.svg" />
      </div>
      <h2 class="warning-title title is-title is-txtDarkBlue is-bold">
        {{ $t('TrialExpired.ExpirationMessage.title') }}
      </h2>
      <p class="text is-body is-txtDarkBlue">
         {{ this.trialExpiredMessage }}
      </p>
      <p class="warning-description text is-body is-txtDarkBlue"
         v-html="$t('TrialExpired.ExpirationMessage.description.phrase2', { expirationDays: this.expirationDays } )">
      </p>
      <div class="buttons-container">
        <a :href="addPaymentsURL" class="button upgrade-button is-caption">
          {{ $t('TrialExpired.ExpirationMessage.actions.addPayments') }}
        </a>
        <a @click="deleteAccount" class="button button--ghost is-txtDarkBlue is-caption">
          {{ $t('TrialExpired.ExpirationMessage.actions.deleteAccount') }}
        </a>
      </div>
    </div>
  </section>
</template>

<script>
import Dialog from 'new-dashboard/components/Backbone/Dialog';
import DeleteAccount from 'new-dashboard/components/Backbone/Dialogs/DeleteAccount.vue';
import format from 'date-fns/format';

export default {
  name: 'ExpirationMessage',
  props: {
    addPaymentsURL: String,
    expirationDays: {
      default: null
    }
  },
  computed: {
    trialExpiredMessage () {
      return (this.hasTrialExpirationDate
        ? this.$t('TrialExpired.ExpirationMessage.description.phrase1.withDate', { expirationDate: this.humanReadableExpirationDate })
        : this.$t('TrialExpired.ExpirationMessage.description.phrase1.noDate'));
    },
    hasTrialExpirationDate () {
      return this.$store.state.user.trial_ends_at;
    },
    humanReadableExpirationDate () {
      return format(this.$store.state.user.trial_ends_at, 'MMMM DD, YYYY');
    }
  },
  methods: {
    deleteAccount () {
      this.$modal.show({
        template: `
        <Dialog @close="$emit('close')">
          <DeleteAccount @close="$emit('close')"/>
        </Dialog>`,
        components: { Dialog, DeleteAccount }
      }, {}, { width: '100%', height: '100%' });
    }
  }
};
</script>

<style scoped lang="scss">
@import 'new-dashboard/styles/variables';

.upgrade-button {
  margin-right: 64px;
  background-color: $button-lockout__bg-color;
}

.link {
  text-decoration: underline;
}

.warning-container {
  text-align: center;
}

.warning-icon {
  margin-bottom: 48px;
}

.warning-title {
  margin-bottom: 16px;
}

.warning-description {
  margin-bottom: 16px;
}

.buttons-container {
  display: flex;
  align-items: center;
  justify-content: center;
  margin-top: 36px;
}
</style>
