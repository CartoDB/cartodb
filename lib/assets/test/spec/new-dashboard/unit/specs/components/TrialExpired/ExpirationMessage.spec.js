import { shallowMount } from '@vue/test-utils';
import ExpirationMessage from 'new-dashboard/components/TrialExpired/ExpirationMessage';

describe('ExpirationMessage.vue', () => {
  it('should render correctly', () => {
    const expirationMessage = createExpirationMessageComponent();
    expect(expirationMessage).toMatchSnapshot();
  });

  describe('Computed Variables', () => {
    it('humanReadableExpirationDate', () => {
      const expirationMessage = createExpirationMessageComponent();
      expect(expirationMessage.vm.humanReadableExpirationDate).toBe('January 10, 2019');
    });
  });

  describe('Methods', () => {
    it('deleteAccount: should open delete account modal', () => {
      const modalMock = {
        show: jest.fn()
      };

      const expirationMessage = createExpirationMessageComponent({
        mocks: { $modal: modalMock }
      });

      expirationMessage.vm.deleteAccount();

      const styleProps = { width: '100%', height: '100%' };
      const componentProps = {
        components: {
          DeleteAccount: expect.any(Object),
          Dialog: expect.any(Object)
        },
        template: `
        <Dialog @close="$emit('close')">
          <DeleteAccount @close="$emit('close')"/>
        </Dialog>`
      };

      expect(modalMock.show).toBeCalledWith(
        componentProps,
        {},
        styleProps
      );
    });
  });
});

function createExpirationMessageComponent (propertiesToOverride = {}) {
  const $store = {
    state: {
      user: { trial_ends_at: '2019-01-10T10:00:00.000Z' }
    }
  };

  const mocks = {
    $t: key => key,
    $store,
    ...propertiesToOverride.mocks
  };

  console.log(mocks);

  return shallowMount(ExpirationMessage, {
    propsData: {
      expirationDays: 10,
      upgradeURL: 'https://user.carto.com/plan'
    },
    ...propertiesToOverride,
    mocks
  });
}
