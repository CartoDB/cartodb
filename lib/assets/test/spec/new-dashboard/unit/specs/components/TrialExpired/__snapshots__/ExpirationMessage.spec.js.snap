// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExpirationMessage.vue should render correctly with trial expired date 1`] = `
<section upgradeurl="https://user.carto.com/plan">
  <div class="container warning-container">
    <div class="warning-icon"><img src="../../assets/icons/common/warning-icon.svg"></div>
    <h2 class="warning-title title is-title is-txtDarkBlue is-bold">
      TrialExpired.ExpirationMessage.title
    </h2>
    <p class="text is-body is-txtDarkBlue">
      TrialExpired.ExpirationMessage.description.phrase1.withDate
    </p>
    <p class="warning-description text is-body is-txtDarkBlue">TrialExpired.ExpirationMessage.description.phrase2</p>
    <div class="buttons-container"><a class="button upgrade-button is-caption">
        TrialExpired.ExpirationMessage.actions.addPayments
      </a> <a class="button button--ghost is-txtDarkBlue is-caption">
        TrialExpired.ExpirationMessage.actions.deleteAccount
      </a></div>
  </div>
</section>
`;

exports[`ExpirationMessage.vue should render correctly without trial expired date 1`] = `
<section upgradeurl="https://user.carto.com/plan">
  <div class="container warning-container">
    <div class="warning-icon"><img src="../../assets/icons/common/warning-icon.svg"></div>
    <h2 class="warning-title title is-title is-txtDarkBlue is-bold">
      TrialExpired.ExpirationMessage.title
    </h2>
    <p class="text is-body is-txtDarkBlue">
      TrialExpired.ExpirationMessage.description.phrase1.noDate
    </p>
    <p class="warning-description text is-body is-txtDarkBlue">TrialExpired.ExpirationMessage.description.phrase2</p>
    <div class="buttons-container"><a class="button upgrade-button is-caption">
        TrialExpired.ExpirationMessage.actions.addPayments
      </a> <a class="button button--ghost is-txtDarkBlue is-caption">
        TrialExpired.ExpirationMessage.actions.deleteAccount
      </a></div>
  </div>
</section>
`;
